File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Servo Bricklet 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var s = new Tinkerforge.BrickletServoV2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Enable position reached callback
        s.setPositionReachedCallbackConfiguration(0, true);

        // Set velocity to 100°/s. This has to be smaller or equal to the
        // maximum velocity of the servo you are using, otherwise the position
        // reached callback will be called too early
        s.setMotionConfiguration(0, 10000, 500000, 500000);
        s.setPosition(0, 9000);
        s.setEnable(0, true);
    }
);

// Register position reached callback
s.on(Tinkerforge.BrickletServoV2.CALLBACK_POSITION_REACHED,
    // Use position reached callback to swing back and forth
    function (servoChannel, position) {
        if(position === 9000) {
            console.log('Position: 90°, going to -90°');
            servo.setPosition(servoChannel, -9000);
        }
        else if(position === -9000) {
            console.log('Position: -90°, going to 90°');
            servo.setPosition(servoChannel, 9000);
        }
        else {
            console.log('Error'); // Can only happen if another program sets position
        }
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        s.setEnable(0, false);

        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/perl/example_callback.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletServoV2;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your Servo Bricklet 2.0

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
our $s = Tinkerforge::BrickletServoV2->new(&UID, $ipcon); # Create device object

# Use position reached callback to swing back and forth
sub cb_position_reached
{
    my ($servo_channel, $position) = @_;

    if ($position == 9000)
    {
        print "Position: 90°, going to -90°\n";
        $s->set_position($servo_channel, -9000);
    }
    elsif ($position == -9000)
    {
        print "Position: -90°, going to 90°\n";
        $s->set_position($servo_channel, 9000);
    }
    else
    {
        print "Error\n"; # Can only happen if another program sets position
    }
}

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register position reached callback to subroutine cb_position_reached
$s->register_callback($s->CALLBACK_POSITION_REACHED, 'cb_position_reached');

# Enable position reached callback
$s->set_position_reached_callback_configuration(0, 1);

# Set velocity to 100°/s. This has to be smaller or equal to the
# maximum velocity of the servo you are using, otherwise the position
# reached callback will be called too early
$s->set_motion_configuration(0, 10000, 500000, 500000);
$s->set_position(0, 9000);
$s->set_enable(0, 1);

print "Press key to exit\n";
<STDIN>;

$s->set_enable(0, 0);

$ipcon->disconnect();
